aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -514,4 +524,17 @@ public void setConf(Configuration conf) {
   protected Map<String, String> properties() {
     return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
   }
+
+  private boolean isTablePurged(TableIdentifier identifier, FileIO tableFileIO) {
```

Comment:
the method name is not really accurate, we are just checking if metadata location is still there, nothing else. What about `tableMetadataExists`